lib/Nessy/Client.pm

```diff
@@ -89,15 +106,17 @@ sub _close_unused_socket {
 }
 
 sub _run_child_process {
-    my($class, %params) = @_;
+    my($self, @sockets) = @_;
+
+    my $params = $self->constructor_params();
 
     eval {
-        $class->_close_unused_socket($params{socketpair}->[PARENT_PROCESS_SOCKET]);
-        my $daemon_class = $class->_daemon_class_name;
+        $self->_close_unused_socket($sockets[PARENT_PROCESS_SOCKET]);
```

Comment:
It's not really part of this PR but I liked the naming convention you used in the test: `daemon_to_client_sock` and `client_to_daemon_sock`.  It might also be an improvement to adopt `client` instead of `parent` and `daemon` instead of `child`.